src/test/kotlin/com/exactpro/th2/infraoperator/integration/IntegrationTest.kt

```diff
@@ -0,0 +1,634 @@
+        kubeClient.createTh2Mstore(TH2_NAMESPACE, MESSAGE_STORAGE_BOX_ALIAS, createAnnotations(gitHash, TEST_CONTENT))
+        kubeClient.awaitHelmRelease(TH2_NAMESPACE, MESSAGE_STORAGE_BOX_ALIAS)
+        // FIXME: estore should have binding
+//        println("Bindings: ${rabbitMQClient.getQueueBindings(RABBIT_MQ_V_HOST, createEstoreQueue(TH2_NAMESPACE))}")
+    }
+
+    @Test
+    @Timeout(30_000)
+    fun `add estore`() {
+        val gitHash = RESOURCE_GIT_HASH_COUNTER.incrementAndGet().toString()
+        kubeClient.createTh2Estore(TH2_NAMESPACE, EVENT_STORAGE_BOX_ALIAS, createAnnotations(gitHash, TEST_CONTENT))
+        kubeClient.awaitHelmRelease(TH2_NAMESPACE, EVENT_STORAGE_BOX_ALIAS)
+        rabbitMQClient.assertBindings(
+            createEstoreQueue(TH2_NAMESPACE),
+            RABBIT_MQ_V_HOST,
+            setOf(
+                "link[$TH2_NAMESPACE:$EVENT_STORAGE_BOX_ALIAS:$EVENT_STORAGE_PIN_ALIAS]",
+            )
+        )
+    }
+
+    /**
+     * Max name length is limited by [HelmRelease.NAME_LENGTH_LIMIT]
+     */
+    @Timeout(30_000)
+    @ParameterizedTest
+    @ValueSource(strings = ["th2-core-component", "th2-core-component-more-than-$NAME_LENGTH_LIMIT-character"])
+    fun `add core component (min configuration)`(name: String) {
+        val gitHash = RESOURCE_GIT_HASH_COUNTER.incrementAndGet().toString()
+        val spec = """
+                imageName: $IMAGE
+                imageVersion: $VERSION
+                type: th2-rpt-data-provider
+        """.trimIndent()
+
+        kubeClient.createTh2CoreBox(
+            TH2_NAMESPACE,
+            name,
+            createAnnotations(gitHash, spec.hashCode().toString()),
+            spec
+        )
+        kubeClient.awaitHelmRelease(TH2_NAMESPACE, hashNameIfNeeded(name)).assertMinCfg(name)
+        rabbitMQClient.assertBindings(
+            createEstoreQueue(TH2_NAMESPACE),
+            RABBIT_MQ_V_HOST,
+            setOf(
+                "link[$TH2_NAMESPACE:$EVENT_STORAGE_BOX_ALIAS:$EVENT_STORAGE_PIN_ALIAS]",
+                "key[$TH2_NAMESPACE:$name:$EVENT_STORAGE_PIN_ALIAS]"
+            )
+        )
+    }
+
+    /**
+     * Max name length is limited by [HelmRelease.NAME_LENGTH_LIMIT]
+     */
+    @Timeout(30_000)
+    @ParameterizedTest
+    @ValueSource(strings = ["th2-component", "th2-component-more-than-$NAME_LENGTH_LIMIT-character"])
+    fun `add component (min configuration)`(name: String) {
+        val gitHash = RESOURCE_GIT_HASH_COUNTER.incrementAndGet().toString()
+        val spec = """
+                imageName: $IMAGE
+                imageVersion: $VERSION
+                type: th2-codec
+        """.trimIndent()
+
+        kubeClient.createTh2Box(
+            TH2_NAMESPACE,
+            name,
+            createAnnotations(gitHash, spec.hashCode().toString()),
+            spec
+        )
+        kubeClient.awaitHelmRelease(TH2_NAMESPACE, hashNameIfNeeded(name)).assertMinCfg(name)
```

Comment:
question: Can we return a resource from `createTh2Box` method and use it instead of the plain name to get the correct name for HelpRelease?